api/jobs.go

```diff
@@ -907,6 +913,7 @@ type Job struct {
 	CreateIndex              *uint64
 	ModifyIndex              *uint64
 	JobModifyIndex           *uint64
+	MaxPriority              *int
```

Comment:
The Job struct doesn't need a `MaxPriority`, as it only ever has a single priority -- the one the user has set or the default.

Reply:
:thinking: I'm not sure how we can send the user defined MaxPriority in the `Job` struct definied in `nomad/structs/structs.go` without touching the one in the api/jobs.go.
My idea was to pass this information so we could update the check at line `nomad/structs/structs.go:4454`. But maybe we can move this in another part, so we don't need the MaxPriority value inside the job struct?

Reply:
Apologies, I'd given a very high-level read and then didn't communicate well. Because scheduling workers are shared between all jobs, the maximum priority value has to be shared between all jobs and can't be configured on a per-job basis. The server config is the maximum value of the priority field.

So the `structs.Job` struct doesn't need a user-defined max priority either. The check that we have in the `(*Job).Validate` method should be checking against the server configured max priority, not a job-internal maximum. So something like the following:

```diff
-if j.Priority < JobMinPriority || j.Priority > JobMaxPriority {
+if j.Priority < JobMinPriority || j.Priority > srv.Server.config.JobMaxPriority {
```

Note that the server config isn't currently exposed to the `Validate` method, so you'll need to thread that through.

Reply:
No worries, thank you for this deeper explanation! :raised_hands: 
In the previous draft, I leaked the JobMaxPriority config from the server directly in the new struct field. 
With your suggestion, the cleanest way I could come up with was to move this check directly into the `jobValidate.Validate` (nomad/job_endpoint_hooks.go:264).
Do you see any downside to this approach? :thinking: 

(I did an amend with the new solution to the previous commit)